### PR TITLE
Problem: zproject-generated files are outdated

### PR DESCRIPTION
**Pull request by gabm**
Solution: update these files to include the latest changes of zproject, mainly the cmake pkg config support

### CODE REVIEW

File: bindings/jni/src/main/java/org/zeromq/czmq/Zconfig.java [hunk 1]
```diff
@@ -238,10 +238,9 @@ public void removeSubtree () {
     }
     /*
     Destroy node and subtree (all children)
-    WARNING: manually fixed void -> long
```
Comment (bluca) [inline bindings/jni/src/main/java/org/zeromq/czmq/Zconfig.java]: please revert changes to this file

Comment (gabm) [reply bindings/jni/src/main/java/org/zeromq/czmq/Zconfig.java]: of course - although its generated by zproject automatically - so maybe we need to fix it there?

Comment (bluca) [reply bindings/jni/src/main/java/org/zeromq/czmq/Zconfig.java]: yes - there's an outstanding issue, but nobody has provided a fix until now

**Comment by bluca** [comment]
The cmake build is broken - the new cmake file was not added to EXTRA_DIST in Makefile.am:

```
CMake Error: File /home/travis/build/zeromq/czmq/czmq-4.2.0/builds/cmake/Config.cmake.in does not exist.
CMake Error at /usr/local/cmake-3.9.2/share/cmake-3.9/Modules/CMakePackageConfigHelpers.cmake:312 (configure_file):
  configure_file Problem configuring file
Call Stack (most recent call first):
  CMakeLists.txt:420 (configure_package_config_file)
```

**Comment by gabm** [comment]
oh I see - I intentionally removed that from my zproject commit oO.. I'll fix it there.. sorry

**Comment by gabm** [comment]
I created a PR here: https://github.com/zeromq/zproject/pull/1089

**Comment by gabm** [comment]
... lets see if it works now..

**Pull request merged**